posthog/temporal/batch_exports/s3_batch_export.py

```diff
@@ -487,58 +540,47 @@ async def flush(
         error: Exception | None,
     ):
         if error is not None:
+            if not self.s3_upload:
+                return
             await self.logger.adebug("Error while writing part %d", self.s3_upload.part_number + 1, exc_info=error)
             await self.logger.awarning(
                 "An error was detected while writing part %d. Partial part will not be uploaded in case it can be retried.",
                 self.s3_upload.part_number + 1,
             )
             return
 
-        await self.logger.adebug(
-            "Uploading part %s containing %s records with size %s bytes",
-            self.s3_upload.part_number + 1,
-            records_since_last_flush,
-            bytes_since_last_flush,
-        )
+        if self.s3_upload is None:
+            self.s3_upload = initialize_upload(self.s3_inputs, self.file_number)
 
-        await self.s3_upload.upload_part(batch_export_file)
+        async with self.s3_upload as s3_upload:
+            await self.logger.adebug(
```

Comment:
If I'm reading this correctly, this implies that each flush corresponds to one S3 upload, or one file. In other words, we cannot upload a single file in multiple flushes, thus the max bytes we flush on must be equal to the file max size (or less if it's the last file). Otherwise we will be triggering flushes before the configured file max size, which would create more smaller files than configured.

This has a potential problem: We must hold in memory/disk all data for the configured file size. Whereas currently we are capped by the 50MB defined as `BATCH_EXPORT_S3_UPLOAD_CHUNK_SIZE_BYTES` parameter, we will now be holding up to 1GB, which is quite the increase.

Moreover, this larger size means we are now waiting longer before updating the heartbeat, thus leading to larger retries (when and if we do crash). 

Finally, this would also means the `BATCH_EXPORT_S3_UPLOAD_CHUNK_SIZE_BYTES` is deprecated. 

I think we should still support multiple parts independently of the configured max file size.